restcomm/restcomm.sms/src/main/java/org/restcomm/connect/sms/smpp/dlr/provider/TelestaxDlrParser.java

```diff
@@ -0,0 +1,97 @@
+package org.restcomm.connect.sms.smpp.dlr.provider;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.log4j.Logger;
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.restcomm.connect.dao.entities.SmsMessage;
+import org.restcomm.connect.dao.entities.SmsMessage.Status;
+import org.restcomm.connect.sms.smpp.dlr.spi.DlrParser;
+
+public class TelestaxDlrParser implements DlrParser {
+    private static final Logger logger = Logger.getLogger(TelestaxDlrParser.class);
+    private static final Map<String, SmsMessage.Status> statusMap;
+    static
+    {
+        statusMap = new HashMap<String, SmsMessage.Status>();
+        statusMap.put("ACCEPTD", SmsMessage.Status.QUEUED);
+        statusMap.put("EXPIRED", SmsMessage.Status.FAILED);
+        statusMap.put("DELETED", SmsMessage.Status.FAILED);
+        statusMap.put("UNDELIV", SmsMessage.Status.FAILED);
+        statusMap.put("REJECTD", SmsMessage.Status.FAILED);
+        statusMap.put("DELIVRD", SmsMessage.Status.DELIVERED);
+        statusMap.put("UNKNOWN", SmsMessage.Status.SENDING);
+    }
+
+    @Override
+    public Map<String, String> parseMessage(String message) {
+        //"id:XXXXXXXXXX sub:001 dlvrd:000 submit date:YYMMDDHHMM done date:YYMMDDHHMM stat:ZZZZZZZ err:YYY text:none
+        Map<String,String> dlr = new HashMap<String, String>();
+
+        final String idTag = "id:";
```

Comment:
can we define these as CONSTANTS? do we have these contants defined in stack in another placem so we can reuse?